samples/extensions/extended_dynamic_state2/README.md

```diff
@@ -0,0 +1,387 @@
+	VkPipelineRenderingCreateInfoKHR pipeline_create{VK_STRUCTURE_TYPE_PIPELINE_RENDERING_CREATE_INFO_KHR};
+	pipeline_create.pNext                   = VK_NULL_HANDLE;
+	pipeline_create.colorAttachmentCount    = 1;
+	pipeline_create.pColorAttachmentFormats = &color_rendering_format;
+	pipeline_create.depthAttachmentFormat   = depth_format;
+	pipeline_create.stencilAttachmentFormat = depth_format;
+
+	/* Skybox pipeline (background cube) */
+	VkSpecializationInfo                    specialization_info;
+	std::array<VkSpecializationMapEntry, 1> specialization_map_entries{};
+	specialization_map_entries[0]        = vkb::initializers::specialization_map_entry(0, 0, sizeof(uint32_t));
+	uint32_t shadertype                  = 0;
+	specialization_info                  = vkb::initializers::specialization_info(1, specialization_map_entries.data(), sizeof(shadertype), &shadertype);
+	shader_stages[0].pSpecializationInfo = &specialization_info;
+	shader_stages[1].pSpecializationInfo = &specialization_info;
+
+	/* Use the pNext to point to the rendering create struct */
+	VkGraphicsPipelineCreateInfo graphics_create{VK_STRUCTURE_TYPE_GRAPHICS_PIPELINE_CREATE_INFO};
+	graphics_create.pNext               = VK_NULL_HANDLE;
+	graphics_create.renderPass          = VK_NULL_HANDLE;
+	graphics_create.pInputAssemblyState = &input_assembly_state;
+	graphics_create.pRasterizationState = &rasterization_state;
+	graphics_create.pColorBlendState    = &color_blend_state;
+	graphics_create.pMultisampleState   = &multisample_state;
+	graphics_create.pViewportState      = &viewport_state;
+	graphics_create.pDepthStencilState  = &depth_stencil_state;
+	graphics_create.pDynamicState       = &dynamic_state;
+	graphics_create.pVertexInputState   = &vertex_input_state;
+	graphics_create.pTessellationState  = VK_NULL_HANDLE;
+	graphics_create.stageCount          = 2;
+	graphics_create.pStages             = shader_stages.data();
+	graphics_create.layout              = pipeline_layouts.baseline;
+
+	
+
+	graphics_create.pNext      = VK_NULL_HANDLE;
+	graphics_create.renderPass = render_pass;
+
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &graphics_create, VK_NULL_HANDLE, &pipeline.baseline));
+...
+```
+
+And now, thanks to dynamic state 2, we can change parameters before each corresponding draw call.
+
+```C++
+	vkCmdBindDescriptorSets(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layouts.baseline, 0, 1, &descriptor_sets.baseline, 0, nullptr);
+	vkCmdBindPipeline(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline.baseline);
+
+	vkCmdSetPrimitiveTopologyEXT(draw_cmd_buffers[i], VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST);
+	vkCmdSetPrimitiveRestartEnableEXT(draw_cmd_buffers[i], VK_FALSE);
+	draw_from_scene(draw_cmd_buffers[i], &scene_nodes, SCENE_BASELINE_OBJ_INDEX);
+
+	vkCmdSetDepthBiasEnableEXT(draw_cmd_buffers[i], VK_FALSE);
+	vkCmdSetRasterizerDiscardEnableEXT(draw_cmd_buffers[i], VK_FALSE);
```

Comment:
Delete 354 and 355 lines (in code this will be inside draw_from_scene function).

Reply:
Done